python/console/console_editor.py

```diff
@@ -739,23 +751,23 @@ def _removeTab(self, tab, tab2index=False):
             if res == QMessageBox.Cancel:
                 return
             if res == QMessageBox.Save:
-                tabWidget.save()
-            if tabWidget.path:
-                self.parent.updateTabListScript(tabWidget.path, action='remove')
+                editorTab.save()
+            if editorTab.path:
+                self.parent.updateTabListScript(editorTab.path, action='remove')
             self.removeTab(tab)
             if self.count() < 1:
                 self.newTabEditor()
         else:
-            if tabWidget.path:
-                self.parent.updateTabListScript(tabWidget.path, action='remove')
+            if editorTab.path:
+                self.parent.updateTabListScript(editorTab.path, action='remove')
             if self.count() <= 1:
                 self.removeTab(tab)
                 self.newTabEditor()
             else:
                 self.removeTab(tab)
 
-        tabWidget.deleteLater()
-        self.currentWidget().editor.setFocus(Qt.TabFocusReason)
+        editorTab.deleteLater()
+        self.currentWidget().setFocus(Qt.TabFocusReason)
```

Comment:
Won't these have slightly different results now? The setFocus will be caught by the widget, not the editor child.

Reply:
Thanks for pointing it out, this prevents the editor from getting the focus when a tab is closed.